server.js

```diff
@@ -26,7 +80,94 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.status(200).json({
+    Message: "Top 50 songs - data",
+    data: listEndpoints(app)
+  });
+});
+
+app.get("/songs", async (req, res) => {
+  const songs = await Song.find().sort({id: 1});
+  res.status(200).json({
+    success: true,
+    message: "OK",
+    body: {
+      topMusicData: songs
+    }
+  });
+});
+
+app.get("/songsOnlyTitleAndArtist", async (req, res) => {
+  const songsOnlyTitleAndArtist = await Song.find().sort({id: 1}).select({trackName: 1, artistName: 1})
+  res.status(200).json({
+    success: true,
+    message: "OK",
+    body: {
+      topMusicData: songsOnlyTitleAndArtist
+    }
+  });
+});
+
+app.get('/song/id/:id', async (req, res) => {
+  try {
+    const songById = await Song.findById(req.params.id);
+    if (songById) {
+      res.status(200).json({
+        success: true,
+        message: "OK",
+        body: {
+          topMusicData: songById
+        }
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        message: "Song Not found",
+        body: {
+          topMusicData: {}
+        }
+      });
+    }
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      message: "Invalid id",
+      body: {
+        topMusicData: {}
+      }
+    });
+  }
+})
+
+app.get("/songs/bpm/:bpm", async (req, res) => {
+  try {
+    const songByBpm = await Song.find({ bpm: req.params.bpm });
+    if (songByBpm) {
+      res.status(200).json({
+        succes: true,
+        message: "OK",
+        body: {
+          topMusicData: songByBpm
+        }
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        message: "Track not found",
+        body: {
+          topMusicData: {}
+        }
+      });
+    }
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      message: "Invalid bmp",
+      body: {
+        topMusicData: {}
+      }
+    });
+  }
 });
```

Comment:
This endpoint also works well to show the correct bpm, but the error did not show up when I entered bpm number that does not exist in the data. One way you can do is to use .length. So for example;

> if (songByBpm.length !== 0)

That way, it will only return data that has length more than zero and also only the ones that match the data. If there is no length of data that does not match, it will return your error message.